### PR TITLE
iEarn financial opportunity

### PR DESCRIPTION
**Pull request by Marik-D**

### CODE REVIEW

File: contracts/TrueReward/IEarnFinancialOpportunity.sol [hunk 1]
```diff
@@ -1,4 +1,84 @@
...
+
+    function configure(
+        yTrueUSDInterface _yToken,
+        TrueRewardBackedToken _token
+    ) public onlyOwner {
+        yToken = _yToken;
+        token = _token;
+    }
+
+    function proxyOwner() public view returns(address) {
```
Comment (wjmelements) [inline contracts/TrueReward/IEarnFinancialOpportunity.sol]: This won't work

File: contracts/TrueReward/IEarnFinancialOpportunity.sol [hunk 2]
```diff
@@ -1,4 +1,84 @@
...
+        return OwnedUpgradeabilityProxy(address(this)).proxyOwner();
```
Comment (wjmelements) [inline contracts/TrueReward/IEarnFinancialOpportunity.sol]: Also, you shouldn't staticcall into yourself if you don't have to; it's gas-intensive

File: contracts/TrueReward/IEarnFinancialOpportunity.sol [hunk 3]
```diff
@@ -1,4 +1,84 @@
...
+    }
+
+    function deposit(address _account, uint256 _amount) external returns(uint256) {
+        require(token.transferFrom(_account, address(this), _amount), "transfer from failed");
+        require(token.approve(address(yToken), _amount), "approve failed");
+        
+        uint256 balanceBefore = yToken.balanceOf(address(this));
+        yToken.deposit(_amount);
```
Comment (wjmelements) [inline contracts/TrueReward/IEarnFinancialOpportunity.sol]: it would surprise me if this did not return the number of shared minted

Comment (terryli0095) [reply contracts/TrueReward/IEarnFinancialOpportunity.sol]: it does not

Comment (terryli0095) [reply contracts/TrueReward/IEarnFinancialOpportunity.sol]: we can do the calculation on our end, but calling balance of is potentially safer

**Pull request merged**